### ISSUE TITLE
Improved error handling and logging

### ISSUE THREAD

**Issue by petargyurov** [petargyurov/megadetector-gui#13]
**Feature**
Currently not all errors are handled. We need a robust system that 1) notifies the user of any errors 2) logs detailed error information to a file

**Additional Detail**
Some errors are already handled, but there isn't much handling of errors coming from the backend process, especially during initiating of the child process.

**Comment by petargyurov** [comment]
The current implemenation is not as robust or as comprehensive as it should be but we have:
- handling local JS errors
- handling backend errors
- handling stdout and stderr streams from the backend
- logging everything to a log file

There is one (maybe two?) major unhandled cases:
- remote code errors (i.e.: errors in libraries)
    -  these cannot be read by JS because of a Chromium CORS security settings (see #15)
- Electron errors in the main process
    - have not put in the time to implement this


I was really hoping that Electron would just have a global error handler that receives a callback but there doesn't seem to be such a feature. On top of that, not all JS errors trigger the typical error event handler... 😡 

Bottom line is, there is still major room for improvement on this front. **One thing that we need to seriously consider is the recycling of the log file.** Right now it will grow in size until a new install (?) erases it.
